lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -189,7 +189,12 @@ def store(payment_method, options = {})
 
         # If customer option is provided, create a payment method and attach to customer id
         # Otherwise, create a customer, then attach
-        if payment_method.is_a?(StripePaymentToken) || payment_method.is_a?(ActiveMerchant::Billing::CreditCard)
+        card_source_pay = payment_method.source.to_s if defined?(payment_method.source)
+        card_brand_pay = card_brand(payment_method) unless payment_method.is_a?(String) || payment_method.nil?
```

Comment:
I don't think this final `payment_method.nil?` check is necessary at this point...